go/arrow/array/float16_builder.go

```diff
@@ -163,3 +168,59 @@ func (b *Float16Builder) newData() (data *Data) {
 
 	return
 }
+
+func (b *Float16Builder) unmarshalOne(dec *json.Decoder) error {
+	t, err := dec.Token()
+	if err != nil {
+		return err
+	}
+
+	switch v := t.(type) {
+	case float64:
+		b.Append(float16.New(float32(v)))
+	case string:
+		f, err := strconv.ParseFloat(v, 32)
+		if err != nil {
+			return err
+		}
+		b.Append(float16.New(float32(f)))
```

Comment:
Does this silently truncate or does it throw an error?

Reply:
Currently it will silently truncate

Reply:
comment added to document the truncation

Reply:
OK, we should probably also make this clear in public docs.

Reply:
It's in the docstring for the `UnmarshalJSON` function which means it'll appear in the public docs for the `UnmarshalJSON` on `Float16Builder`